test-case/check-reboot.sh

```diff
@@ -112,17 +114,23 @@ else
     full_cmd=${full_cmd#\/bin\/bash}
 fi
 
+# some load will use '~' which is $HOME, but after system reboot, in rc.local $USER is root
+# so the '~' will lead to the error path
+full_cmd=$(echo $full_cmd|sed "s:~:$HOME:g")
+
 # load script default value for the really full command
-if [ $# -eq 0 ]; then
-    full_cmd=$(echo $full_cmd|sed "s:$0:& -l $loop_count:g")
-fi
+# add -d delay
+[[ ! $(echo $full_cmd|grep "'-d $delay'") ]] && full_cmd=$(echo $full_cmd|sed "s:$0:& -d $delay:g")
+# add -t timeout
+[[ ! $(echo $full_cmd|grep "'-t $timeout'") ]] && full_cmd=$(echo $full_cmd|sed "s:$0:& -t $timeout:g")
+# add -l loop_count
+[[ ! $(echo $full_cmd|grep "'-l $loop_count'") ]] && full_cmd=$(echo $full_cmd|sed "s:$0:& -l $loop_count:g")
+# now command will like: $0 -l loop_count -t timeout -d delay
 
 # convert relative path to absolute path
 full_cmd=$(echo $full_cmd|sed "s:$0:$(realpath $0):g")
```

Comment:
Is this duplicated with 
```
# some load will use '~' which is $HOME, but after system reboot, in rc.local $USER is root
# so the '~' will lead to the error path
full_cmd=$(echo $full_cmd|sed "s:~:$HOME:g")
```

Reply:
it is not duplicated, because use different load  
1. load as "./check-reboot.sh" or "xxx/check-reboot.sh"
2. if without this change, it will cause mistake, in bash script, "~" is absolute path ..

Reply:
> Most part looks good.
> One mirror question inline and could we just make all cmd parse in one place and use a wrapper function for it

which part? I don't think one line command need to add the function, which will been hard to understand  
on the other hand, just reboot need to convert this information, so let keep it  
if have other case need to fix full command, I will add the new function for it